### PR TITLE
Add documentation for '[]'

### PR DESCRIPTION
**Pull request by Bestra**
Addresses #2273. 
- Document `[]` before talking about `[]`
- Shuffles the `@each` heading to be consistent with the new `[]
- reorders the circumstances for busting the CP cache from least specific
  to most specific

### CODE REVIEW

**Comment by jenweber** [comment]
@locks can you review this one? Thanks!

File: source/object-model/computed-properties-and-aggregate-data.md [hunk 1]
```diff
@@ -1,9 +1,52 @@
+When a computed property depends on the contents of an array, there are a few
+extra methods you'll need to use in order to correctly recognize when the
+contents of the array change. Arrays have two special keys you can append to
+array properties to track changes on them, `[]` and `@each`.
+
+## `[]`
+
+Sometimes a computed property needs to update when items are added or removed from an array.
```
Comment (locks) [inline source/object-model/computed-properties-and-aggregate-data.md]: It should also trigger if you replace one of the items (like, when you put a value directly in an index).

File: source/object-model/computed-properties-and-aggregate-data.md [hunk 2]
```diff
@@ -1,9 +1,52 @@
...
+In those cases we can use the `[]` array key to tell the property to update at the right time.
+We'll use the obligatory todo list for our examples:
```
Comment (locks) [inline source/object-model/computed-properties-and-aggregate-data.md]: Could you replace obligatory with familiar?

Comment (locks) [inline source/object-model/computed-properties-and-aggregate-data.md]: Can you replace obligatory with familiar?

File: source/object-model/computed-properties-and-aggregate-data.md (hunk 1 above)
Comment (Bestra) [reply source/object-model/computed-properties-and-aggregate-data.md]: You're right!

```
Sometimes you need a computed property to update when you add, remove, or replace an item in an array.
```
^^ Do you think that's too concise?

I think it'd also be worth calling the replacement case out explicitly in the two lists for `[]` and `@each` as well. 

I'll make a new commit with a crack at the changes and squash once we settle on the final wording

Comment (Bestra) [reply source/object-model/computed-properties-and-aggregate-data.md]: Alternate wording
```
Sometimes you need a computed property to update when the membership of an array changes.
```

**Pull request merged**